### ISSUE TITLE
Wrong Values in Reports when using various currencies

### ISSUE THREAD

**Issue by bgvr0607** [moneymanagerex/android-money-manager-ex#1620]
**Describe the bug**
 I kept using an old version of the Android App which supported syncing (2018.11.15.987) since everything worked well for me. Recently the sync stopped working, so I had to upgrade the app. With newer versions for me the numbers shown in reports income vs. expenses are sometimes different between the Android app and the desktop app. This seem to be linked to usage of different currencies. Looking at the numbers, the exchange rate does not seem to be applied correctly. Using an exchange rate of 1 for all entries with different currencies matches the numbers I see.

Testing different versions, I found that until version 2019.02.14.994 the numbers were shown correctly.

**Expected behavior**
Using the same database for Android and Desktop Apps should display the same number.

**Device Information:**
 - Device: OnePlus 8T
 - OS: OxygenOS 13.1 (Android 13)
 - Storage: Sync to Dropbox
 - Encrypted: No
 - App Version: every version above 2019.02.14.994

**Comment by guanlisheng** [comment]
hi @bgvr0607, thanks for using mmex and reporting the bug.

we just addressed this bug and can you help to verify the apk via https://drive.google.com/file/d/1adbTNIsGBawEukogULl_Mppo2f2jAhCH/view?usp=sharing

**Comment by bgvr0607** [comment]
Hi @guanlisheng, 

Thank you for working on this topic right away!

I tested the version you shared, the reports are now matched between the Android version and the Desktop app.

**Comment by guanlisheng** [comment]
btw, do you have any best practices/advice on upgrading from very old ammex to latest one , e.g. #1543

**Comment by bgvr0607** [comment]
I checked the recomendation list by jorgeEF, it describes the general workflow I also follow. But there are things I additionally do to see if the new version works for me. Since I also got quite many error messages related to the db, I wanted to confirm everything is as it should be. That is also how I found the inconsistency in the report.

Here the general workflow:
- note down key values, such as balance, with current android app version
- follow the recommendation list by jorgeEF
- make 2 new folders on DropBox for experimenting (1x for Android App, 1x for Desktop App) - I read somewhere that using Android and Desktop on the same db file used to cause issues in the past. But since both now use the same db version that should be solved, right?
- copy the db to both folders
- load db with new android app and with desktop app from the respective folders
- compare if things are consistent between apps and that the values are the same as before (and there are no persistent db errors)

I did not use a newer version besides testing, due to wrong values in the reports. Later I will try upgrading and if I find anything additional I will let you know.
